src/main/kotlin/eu/europa/ec/eudi/openid4vci/Issuance.kt

```diff
@@ -141,27 +141,52 @@ sealed interface IssuanceRequestPayload {
      */
     data class ConfigurationBased(
         val credentialConfigurationIdentifier: CredentialConfigurationIdentifier,
-        val claimSet: ClaimSet?,
+        val claimSet: ClaimSet? = null,
     ) : IssuanceRequestPayload
 }
 
 typealias AuthorizedRequestAnd<T> = Pair<AuthorizedRequest, T>
 
 /**
- * An interface for submitting a credential issuance request. Contains all the operation available to transition an [AuthorizedRequest]
- * to a [SubmissionOutcome]
+ * An interface for submitting a credential issuance request.
  */
 interface RequestIssuance {
 
+    /**
+     * Places a request to the credential issuance endpoint.
+     * Method will attempt to automatically retry submission in case
+     * - Initial authorization state is [AuthorizedRequest.NoProofRequired] and
+     * - a [popSigner] has been provided
+     *
+     * @receiver the current authorization state
+     * @param requestPayload the payload of the request
+     * @param popSigner Signer component of the proof to be sent. Although this is an optional
+     * parameter, only required in case the present authorization state is [AuthorizedRequest.ProofRequired],
+     * caller is advised to provide it, in order to allow the method to automatically retry
+     * in case of [SubmissionOutcome.InvalidProof]
+     *
+     * @return the possibly updated [AuthorizedRequest] (if updated it will contain a fresh c_nonce) and
+     * the [SubmissionOutcome]
+     */
+    suspend fun AuthorizedRequest.requestSingleAndUpdateState(
+        requestPayload: IssuanceRequestPayload,
+        popSigner: PopSigner?,
+    ): Result<AuthorizedRequestAnd<SubmissionOutcome>>
+
     /**
      *  Requests the issuance of a single credential having an [AuthorizedRequest.NoProofRequired] authorization.
      *
      *  @param requestPayload   The payload of the request.
      *  @return The new state of the request or error.
      */
+    @Deprecated(
```

Comment:
I deprecated mainly because I don't like the return type which is just `SubmissionOutcome` 
The correct is `AuthorizedRequestAnd<SubmissionOutcome>`

What you say about `AuthorizedRequest` is totally true. But if we go this way we should also remove `c_nonce` from `SubmissionOutcome`.  
Had implemented that, but didn't commit it.

Perhaps we can open a new issue to simplify things:
- Make AuthorizedRequest a simple data class 
- Remove from SubmissionOutcome cNonce

The implementation of both it would be straight forward 

Introduce inside RequestIssuanceImpl two private sealed interfaces as they are now the public.
Let all private methods RequestIssuanceImpl to use AuthorizedRequestInternal and SubmissionOutcomeInternal
At the end just map to the public